modules/TransitionMixin.js

```diff
@@ -0,0 +1,157 @@
+var warning = require('react/lib/warning');
+var { getParamNames } = require('./PathUtils');
+
+function forEachComponent(components, callback) {
+  if (Array.isArray(components)) {
+    components.forEach(callback);
+  } else if (typeof components === 'object') {
+    for (var key in components)
+      if (components.hasOwnProperty(key))
```

Comment:
that's what `components.hasOwnProperty(key)` protects against. If its a key on the prototype, it will be false and ignored.